--generate-ctest-testcase: generated test case does not compile when the counterexample has a nondet pointer

## Summary

`--generate-ctest-testcase` emits a `test_case.cpp` that does not compile whenever the counterexample contains a nondeterministic pointer. The generated `__VERIFIER_nondet_pointer` stub returns a `const void*` from a function declared to return `void*`.

This is ill-formed in C++ (hard error) and a constraint violation in C (GCC diagnoses it as a warning by default, an error under `-pedantic-errors`). Since the flag's whole purpose is to produce a buildable, runnable test case, the affected counterexamples yield no usable artefact.

## Reproducer

```cpp
// nondet_pointer_testcase.cpp
#include <cassert>

extern "C" void *__VERIFIER_nondet_pointer(void);

int main()
{
  void *p = __VERIFIER_nondet_pointer();
  assert(p != nullptr);
  return 0;
}
```

```
$ esbmc --std c++11 nondet_pointer_testcase.cpp --generate-ctest-testcase
...
VERIFICATION FAILED
```

The generated `test_case.cpp` contains:

```cpp
void* __VERIFIER_nondet_pointer(void) {
  static int i = 0;
  static const void* v[] = { 0 };
  return v[i++];
}
```

Following the documented flow from https://esbmc.github.io/docs/c-cpp/ctest-gen/:

```
$ mkdir build && cd build && cmake .. && cmake --build .
test_case.cpp:13:15: error: invalid conversion from 'const void*' to 'void*' [-fpermissive]
   13 |   return v[i++];
      |          ~~~~~^
gmake: *** [Makefile:101: all] Error 2
```

In C the same body is accepted with a warning by default, but rejected under `-pedantic-errors`:

```
error: return discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
```

## Cause

`write_nondet_functions` (`src/goto-symex/ctest.cpp:378`) writes the array as

```cpp
out << "  static const " << c_type << " v[] = { ";
```

For pointer types `type_to_c_string` (`src/goto-symex/ctest.cpp:167`) returns `"void*"`, so `const` + `void*` textually composes to `const void* v[]` — an array of *pointer to const void* — rather than a const array of `void*`. The subsequent `return v[i++];` then drops a qualifier.

The `const` placement is correct for every other type; only the pointer case has the qualifier bind to the pointee.

## Suggested fix

Emit the pointer array as `static void *const v[]` (const array of `void*`), or special-case the pointer branch to declare `static void * v[]`. A minimal patch in `write_nondet_functions` that swaps the declarator when `c_type` ends in `*` would cover it.

## Impact

Any target whose counterexample includes a nondet pointer. In our case (verifying `Aws::Utils::Base64::Base64::Decode` from aws-sdk-cpp) both counterexamples were affected, and the generated test cases had to be rewritten with `sed` before they could be built.

## Version

ESBMC 8.4.0, master `d0bb9881f2`, x86_64 Linux.

--- a/regression/witnesses/test_case_generation/ctest_gen_pointer/main.c
+++ b/regression/witnesses/test_case_generation/ctest_gen_pointer/main.c
@@ -1,0 +1,10 @@
+#include <assert.h>
+
+void *__VERIFIER_nondet_pointer(void);
+
+int main()
+{
+  void *p = __VERIFIER_nondet_pointer();
+  assert(p != 0);
+  return 0;
+}

--- a/regression/witnesses/test_case_generation/ctest_gen_pointer/test.desc
+++ b/regression/witnesses/test_case_generation/ctest_gen_pointer/test.desc
@@ -1,0 +1,7 @@
+CORE
+main.c
+--generate-ctest-testcase
+^Generated CTest test case\: test_case\.c$
+^VERIFICATION FAILED$
+CHECK_FILE test_case.c contains static void\* const v\[\]
+CHECK_FILE test_case.c absent static const void\*

--- a/regression/witnesses/test_case_generation/ctest_gen_pointer_cpp/main.cpp
+++ b/regression/witnesses/test_case_generation/ctest_gen_pointer_cpp/main.cpp
@@ -1,0 +1,10 @@
+#include <cassert>
+
+extern "C" void *__VERIFIER_nondet_pointer(void);
+
+int main()
+{
+  void *p = __VERIFIER_nondet_pointer();
+  assert(p != nullptr);
+  return 0;
+}

--- a/regression/witnesses/test_case_generation/ctest_gen_pointer_cpp/test.desc
+++ b/regression/witnesses/test_case_generation/ctest_gen_pointer_cpp/test.desc
@@ -1,0 +1,7 @@
+CORE
+main.cpp
+--std c++11 --generate-ctest-testcase
+^Generated CTest test case\: test_case\.cpp$
+^VERIFICATION FAILED$
+CHECK_FILE test_case.cpp contains static void\* const v\[\]
+CHECK_FILE test_case.cpp absent static const void\*

--- a/src/goto-symex/ctest.cpp
+++ b/src/goto-symex/ctest.cpp
@@ -375,7 +375,9 @@ static void write_nondet_functions(
 
     out << c_type << " __VERIFIER_nondet_" << verifier_type << "(void) {\n";
     out << "  static int i = 0;\n";
-    out << "  static const " << c_type << " v[] = { ";
+    // East const: c_type is "void*" for pointers, so a leading const would
+    // qualify the pointee and make `return v[i++]` discard it.
+    out << "  static " << c_type << " const v[] = { ";
     for (size_t j = 0; j < values.size(); ++j)
     {
       if (j > 0)

